service/controller/resource/spark/create.go

```diff
@@ -141,8 +140,8 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 
 	{
-		if !bytes.Equal(ignitionBlob, dataSecret.Data[ignitionBlobKey]) {
-			dataSecret.Data[ignitionBlobKey] = ignitionBlob
+		if !bytes.Equal(ignitionBlob, dataSecret.Data[key.CloudConfigSecretKey]) {
+			dataSecret.Data[key.CloudConfigSecretKey] = ignitionBlob
```

Comment:
I moved this to the `key` package because we need it when writing but also when reading the value from the secret.